### PR TITLE
bambootracker: 0.6.0 -> 0.6.1

### PR DESCRIPTION
**Pull request by OPNA2608**
###### Description of changes
Struggling with Darwin:

1. All Qt5 applications are currently failing to launch on my system.
```
qt.qpa.plugin: Could not load the Qt platform plugin "cocoa" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: cocoa, minimal, offscreen.

Abort trap: 6
```

2. I tried to migrate this to `stdenv.mkDerivation` which would also allow a Qt6 variant, but that causes the build to fail:
```
In file included from audio/audio_stream_rtaudio.cpp:29:
In file included from /nix/store/b1d9agnhl0p5fp298yz0hkxx9s75rnlf-rtaudio-5.2.0/include/rtaudio/RtAudio.h:877:
In file included from /nix/store/nzg19z5bqikz64k8z6wy16yia0wc7p0l-apple-framework-CoreAudio-11.0.0/Library/Frameworks/CoreAudio.framework/Headers/AudioHardware.h:86:
In file included from /nix/store/57giqsy5k5gcrz7gvyaik91izgj0n3z1-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:60:
In file included from /nix/store/57giqsy5k5gcrz7gvyaik91izgj0n3z1-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFPropertyList.h:17:
In file included from /nix/store/57giqsy5k5gcrz7gvyaik91izgj0n3z1-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFStream.h:16:
/nix/store/57giqsy5k5gcrz7gvyaik91izgj0n3z1-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFURL.h:1189:83: error: expected ','
    kCFURLBookmarkCreationWithSecurityScope API_AVAILABLE(macos(10.7), macCatalyst(13.0))  API_UNAVAILABLE(ios, watchos, tvos) = ( 1UL << 11 ), // Mac OS X 10.7.3 and later, include information in the bookmark data which allows the same sandboxed process to access the resource after being relaunched
                                                                                  ^
/nix/store/57giqsy5k5gcrz7gvyaik91izgj0n3z1-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFURL.h:1190:98: error: expected ','
    kCFURLBookmarkCreationSecurityScopeAllowOnlyReadAccess API_AVAILABLE(macos(10.7), macCatalyst(13.0)) API_UNAVAILABLE(ios, watchos, tvos) = ( 1UL << 12 ), // Mac OS X 10.7.3 and later, if used with kCFURLBookmarkCreationWithSecurityScope, at resolution time only read access to the resource will be granted
                                                                                                 ^
/nix/store/57giqsy5k5gcrz7gvyaik91izgj0n3z1-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFURL.h:1200:85: error: expected ','
    kCFURLBookmarkResolutionWithSecurityScope API_AVAILABLE(macos(10.7), macCatalyst(13.0)) API_UNAVAILABLE(ios, watchos, tvos) = ( 1UL << 10 ), // Mac OS X 10.7.3 and later, use the secure information included at creation time to provide the ability to access the resource in a sandboxed process.
                                                                                    ^
3 errors generated.
make[1]: *** [Makefile:4186: audio_stream_rtaudio.o] Error 1
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/bambootracker/default.nix [hunk 1]
```diff
@@ -2,38 +2,52 @@
...
 
-  qmakeFlags = [ "CONFIG+=system_rtaudio" "CONFIG+=system_rtmidi" ];
+  qmakeFlags = [
+    "CONFIG+=system_rtaudio"
+    "CONFIG+=system_rtmidi"
+  ];
 
   postConfigure = "make qmake_all";
 
+  dontWrapQtApps = stdenv.hostPlatform.isDarwin;
```
Comment (wegank) [inline pkgs/applications/audio/bambootracker/default.nix:44]:
```suggestion
```

File: pkgs/applications/audio/bambootracker/default.nix [hunk 2]
```diff
@@ -2,38 +2,52 @@
...
-    sha256 = "sha256-yubaKTc8NFLxMY0/5c2VubRHgAGOsRlitmXJ1UHzl60=";
+    hash = "sha256-Ymi1tjJCgStF0Rtseelq/YuTtBs2PrbF898TlbjyYUw=";
   };
 
-  nativeBuildInputs = [ qmake qttools pkg-config ];
+  nativeBuildInputs = [
+    pkg-config
+    qmake
+    qttools
+  ];
```
Comment (wegank) [inline pkgs/applications/audio/bambootracker/default.nix:29]:
```suggestion
  nativeBuildInputs = [
    pkg-config
    qmake
    qttools
    wrapQtAppsHook
  ];
```

File: pkgs/applications/audio/bambootracker/default.nix [hunk 3]
```diff
@@ -2,38 +2,52 @@
 , stdenv
 , lib
 , fetchFromGitHub
-, qmake
 , pkg-config
-, qttools
+, qmake
 , qtbase
+, qttools
```
Comment (wegank) [inline pkgs/applications/audio/bambootracker/default.nix:8]:
```suggestion
, qttools
, wrapQtAppsHook
```

**Comment by wegank** [comment]
All of the above for the first issue. For the second, you should override stdenv with darwin.apple_sdk_11_0.stdenv, since Qt 5.15.8+ is built with macOS 11 SDK in nixpkgs.

**Comment by OPNA2608** [comment]
`darwin.apple_sdk_11_0.stdenv` hint helps, code suggestions do not: `wrapQtApp` is called manually in `postInstall` on Darwin to wrap the binary inside the app bundle instead of the symlink to it in `bin`. That way both `open -a BambooTracker.app` and `BambooTracker` work. The binary that gets launched is definitely wrapped the same way `wrapQtAppsHook` would do it:

Screenshot_macOS_2023-02-06_17:20:49

I also cannot launch `ptcollab` or `neovim-qt` on master (https://github.com/NixOS/nixpkgs/commit/0406afa8c713de794aa2fb5d42834ec9b2b1e8e0) anymore, same error. My channel is at https://github.com/NixOS/nixpkgs/commit/293a28df6d7ff3dec1e61e37cc4ee6e6c0fb0847 and all 3 applications work there.

**Comment by wegank** [comment]
> I also cannot launch `ptcollab` or `neovim-qt` on master ([0406afa](https://github.com/NixOS/nixpkgs/commit/0406afa8c713de794aa2fb5d42834ec9b2b1e8e0)) anymore, same error. My channel is at [293a28d](https://github.com/NixOS/nixpkgs/commit/293a28df6d7ff3dec1e61e37cc4ee6e6c0fb0847) and all 3 applications work there.

The regression came from #210062, which was fairly recent. I'm also having build failures in other packages, and I once thought of overriding stdenv globally in Qt 5, as was done for Qt 6, but I'm not able to rebuild everything on my own machine.

**Comment by OPNA2608** [comment]
We'll assume that issue 1 will get fixed treewide in the future then.

Also pushed a Qt6 variant now, works on x86_64-linux but I'm unable to test it on Darwin because my macOS version is too old to run any Qt6 binaries. Gonna have to leave that one to either ofborg or a reviewer ¯\\\_(ツ)\_/¯
```
configuring
QMAKEPATH=/nix/store/xbsjv054li93p6gzn0f8kpyzcryriznl-qtbase-6.4.2-dev:/nix/store/mx8b39jf0rgxnnwrcyc2hiyqzyqlb88r-qttools-6.4.2-dev:/nix/store/a0zyddp0g8bb895kxic38mxq54avaa97-qtdeclarative-6.4.2-dev:/nix/store/lm6c0ffvkbn1q5xps1i0ksxl0s2qvkpf-qtshadertools-6.4.2-dev:/nix/store/6zkmf6j37q6xn23gz3gcaw895048z52p-qt5compat-6.4.2-dev
qmake PREFIX=/nix/store/3nmzqbx9gpy7baw562agy4475iplwq4n-bambootracker-0.6.1 NIX_OUTPUT_OUT=/nix/store/3nmzqbx9gpy7baw562agy4475iplwq4n-bambootracker-0.6.1 NIX_OUTPUT_DEV=/nix/store/3nmzqbx9gpy7baw562agy4475iplwq4n-bambootracker-0.6.1 NIX_OUTPUT_BIN=/nix/store/3nmzqbx9gpy7baw562agy4475iplwq4n-bambootracker-0.6.1 NIX_OUTPUT_QML=/nix/store/3nmzqbx9gpy7baw562agy4475iplwq4n-bambootracker-0.6.1/lib/qt-6/qml NIX_OUTPUT_PLUGIN=/nix/store/3nmzqbx9gpy7baw562agy4475iplwq4n-bambootracker-0.6.1/lib/qt-6/plugins CONFIG+=release CONFIG+=system_rtaudio CONFIG+=system_rtmidi
dyld: Symbol not found: _NSAppearanceNameDarkAqua
  Referenced from: /nix/store/z154sm1m3g2pfrbfxavwm03a6anadl2f-qtbase-6.4.2/lib/libQt6Core.6.dylib (which was built for Mac OS X 10.14)
  Expected in: /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit
 in /nix/store/z154sm1m3g2pfrbfxavwm03a6anadl2f-qtbase-6.4.2/lib/libQt6Core.6.dylib
/nix/store/hkyidc6gli1s55vhs7xhhdhq29igq8zf-qmake6-hook/nix-support/setup-hook: line 31:  2139 Abort trap: 6           qmake "${qmakeFlags[@]}"
```

**Comment by wegank** [comment]
I have several commits on master overwritten by the last staging-next merge, so qmake is currently broken again on Qt 6. I'll try to fix that now, and I'll come back to you soon.

**Comment by wegank** [comment]
@ofborg build bambootracker bambootracker-qt6 bambootracker-qt6.passthru.tests bambootracker.passthru.tests

**Review by wegank — APPROVED** [review]

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 214913` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>bambootracker</li>
    <li>bambootracker-qt6</li>
  </ul>
</details>

**Comment by wegank** [comment]
```bash
$ open results/bambootracker-qt6/Applications/BambooTracker.app
```

Capture d’écran 2023-02-07 à 07 10 44

**Pull request merged**